consensus/src/rand/rand_gen/block_queue.rs

```diff
@@ -0,0 +1,248 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::experimental::buffer_manager::OrderedBlocks;
+use aptos_consensus_types::{
+    common::Round,
+    executed_block::ExecutedBlock,
+    randomness::{RandMetadata, Randomness},
+};
+use std::collections::BTreeMap;
+
+/// Maintain the ordered blocks received from consensus and corresponding randomness
+pub struct QueueItem {
+    ordered_blocks: OrderedBlocks,
+    offsets_by_round: BTreeMap<Round, usize>,
```

Comment:
why not HashMap?

Reply:
that's @zjma 's beautiful code

Reply:
when `try_deque`ing we need to iterate in the round order. So hash map won't work

Reply:
You have a BTreeMap in struct BlockQueue to maintain the order, no?

Reply:
either is fine? there won't be a lot of rounds in the same `OrderedBlocks`? The most I saw is 3.

Reply:
(you are right @grao1991 , i deleted my comment earlier.)